docs/branch_protection_rule.md

```diff
@@ -0,0 +1,40 @@
+<!--
+    Copyright 2019 VMware, Inc.
+    SPDX-License-Identifier: BSD-2-Clause
+-->
+
+## Setting up a Branch Protection Rule
+
+
+In order to allow Precaution to prevent pull requests from merging, do the following:
+
+### 1. Create a pull request
+
+You should create a pull request after the installation of Precaution in order to setup Precaution as a branch protection rule.
+
```

Comment:
Is the reason we need to have a PR because the protection rule can only be configured once the status check has been observed for the repository?

Reply:
Even if you had installed an app 1 year ago if you haven't created a pull request Precaution wouldn't be visible in the "Require status checks to pass before merging" option:
![image](https://user-images.githubusercontent.com/16246778/57908760-80e73700-7889-11e9-8a50-98d99999747c.png)

You will have the same situation when you decide to remove Precaution then until you create a pull request Precaution will be visible "Require status checks to pass before merging" option.

Reply:
Thanks for explaining. Can we expand the documentation a bit more to explain why the user needs to create a pull request and why it doesn't have to be merged?

Reply:
I added a paragraph explaining those things.
What do you think?

Reply:
Looks good, minor nits then let's merge it.